spinetoolbox/spine_db_editor/widgets/graph_view_mixin.py

```diff
@@ -181,6 +194,71 @@ def _accepts_entity_item(self, item, db_map):
             return True
         return False
 
+    def _alternative_accepts(self, item, db_map):
+        selected_alternative_ids = self._filter_alternative_ids.get(db_map, set())
+        selected_scen_alts = self._filter_scenario_ids.get("scenario_alternative", {}).get(db_map, set())
+        if not (selected_alternative_ids or selected_scen_alts):
+            if item["id"] in self.highlight_by_id:
+                self.highlight_by_id.pop(item["id"])
+            return True  # No alternatives
+        activities = set()
+        all_alternatives = selected_alternative_ids | selected_scen_alts
+        for alt_id in all_alternatives:
+            entity_alternative = db_map.get_entity_alternative_item(entity_id=item["id"], alternative_id=alt_id)
+            if entity_alternative:
+                if entity_alternative["active"]:
+                    activities.add(True)
+                    continue
+            activities.add(False)
+        if not activities:
+            return False  # No entity alternatives for this guy
+        if all(activities):
+            return True  # Active in all selected alternatives, good to go.
+        if any(activities):
+            self.highlight_by_id.update({item["id"]: EntityBorder.CONFLICTED})
+            return True  # Active in some selected alternatives, good to go.
+        self.highlight_by_id.update({item["id"]: EntityBorder.INACTIVE})
+        return False  # Not active in any selected alternatives, no go.
+
+    def _scenario_accepts(self, item, db_map):
+        scenarios = self._filter_scenario_ids.get("scenario", {}).get(db_map, {})
+        if not scenarios:
+            return True  # No scenarios selected
+        for scenario_id in scenarios:
+            state = db_map.item_active_in_scenario(item, scenario_id)
+            if state is not None:
+                return state
+        for scenario_id in scenarios:
+            state = db_map.item_active_in_scenario(item, scenario_id)
+            if state is not None:
+                return state
+        if item["element_id_list"]:
+            results = set()
+            for id_ in item["element_id_list"]:
+                element = self.db_mngr.get_item(db_map, "entity", id_)
+                results.add(self._scenario_accepts(element, db_map))
+            return all(results)  # Accept only if all elements are accepted also
+        entity_class = self.db_mngr.get_item(db_map, "entity_class", item["class_id"])
+        if entity_class["active_by_default"]:
+            self.highlight_by_id.update({item["id"]: EntityBorder.INACTIVE})
```

Comment:
Again, for simple changes, I would use `self.highlight_by_id[item["id"]] =EntityBorder.INACTIVE`